dl/src/main/scala/com/intel/analytics/bigdl/nn/TimeDistributedCriterion.scala

```diff
@@ -29,77 +29,66 @@ import scala.reflect.ClassTag
  * @param critrn
  */
 
-class TimeDistributedCriterion[T : ClassTag](critrn : TensorCriterion[T])
+class TimeDistributedCriterion[T : ClassTag](critrn : TensorCriterion[T],
+                                             sizeAverage: Boolean = true)
 (implicit ev: TensorNumeric[T]) extends TensorCriterion[T] {
 
-  private val fInput: Tensor[T] = Tensor[T]()
-  private val fTarget: Tensor[T] = Tensor[T]()
-  private var inputSize: Array[Int] = _
-  private var targetSize: Array[Int] = _
-
-  private def combine(src: Array[Int], target: Array[Int]): Unit = {
-    require(src.length == target.length + 1,
-      "TimeDistributedCriterion: combine method requires src.length == target.length + 1" +
-        s" Current src.length = ${src.length}" +
-        s" Current target.length = ${target.length}")
-    target(0) = src(0) * src(1)
-    var j = 1
-    while (j < target.length) {
-      target(j) = src(j + 1)
-      j += 1
-    }
-  }
+  private var fInput: Tensor[T] = Tensor[T]()
+  private var fTarget: Tensor[T] = Tensor[T]()
 
   override def updateOutput(input: Tensor[T], target: Tensor[T]): T = {
     /**
+     * Take each time slice of input and target, and add up all outputs of slices
      * For example
-     * input.size = [B, T, D] => inputSize = [B * T, D]
-     * target.size = [B, T] => targetSize = [B * T]
+     * input.size = [B, T, D] => fInput.size = [B, D]
+     * target.size = [B, T] => fTarget.size = [B]
+     * If sizeAverage is true, the output is averaged through time dimension
      */
```

Comment:
Input dimension is not necessarily 3D since it depends on which criterion is used. But requirement such as input.dim() >= 2 can be added anyway. Is it necessary?